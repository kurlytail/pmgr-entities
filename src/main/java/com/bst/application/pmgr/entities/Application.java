package com.bst.application.pmgr.entities;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.web.servlet.support.SpringBootServletInitializer;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.transaction.annotation.EnableTransactionManagement;

import com.bst.configuration.pmgr.entities.PmgrEntitiesConfiguration;

@SpringBootApplication
@ComponentScan(basePackageClasses = { PmgrEntitiesConfiguration.class, Application.class })
@EnableTransactionManagement
public class Application extends SpringBootServletInitializer {

	public static void main(final String[] args) {
		SpringApplication.run(Application.class, args);
	}
}