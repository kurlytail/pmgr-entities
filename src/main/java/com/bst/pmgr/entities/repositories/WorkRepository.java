package com.bst.pmgr.entities.repositories;

import org.springframework.data.repository.CrudRepository;
import org.springframework.data.rest.core.annotation.RepositoryRestResource;

import com.bst.pmgr.entities.Work;
import com.google.common.base.Optional;

@RepositoryRestResource
public interface WorkRepository extends CrudRepository<Work, Long>, WorkRepositoryStrategy {
	Optional<Work> findByName(final String name);
}
