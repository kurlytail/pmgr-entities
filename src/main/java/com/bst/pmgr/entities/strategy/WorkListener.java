package com.bst.pmgr.entities.strategy;

import java.util.stream.Collectors;

import javax.persistence.PrePersist;

import org.kohsuke.randname.RandomNameGenerator;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import com.bst.pmgr.entities.Activity;
import com.bst.pmgr.entities.Work;
import com.bst.pmgr.entities.schema.SchemaService;

@Component
public class WorkListener {

	private RandomNameGenerator rnd = new RandomNameGenerator(0);

	@Autowired
	private SchemaService schemaService;

	@PrePersist
	public void prePersist(final Work work) throws Exception {
		if (work.getName() == null) {
			work.setName(rnd.next());
		}

		if (work.getActivities().size() == 0) {
			work.getActivities().addAll(schemaService.getActivities().values().stream().map(metaActivity -> {
				Activity activity = new Activity();
				activity.setMetaName(metaActivity.getName());
				activity.setName(rnd.next());
				return activity;
			}).collect(Collectors.toList()));
		}
	}
}
