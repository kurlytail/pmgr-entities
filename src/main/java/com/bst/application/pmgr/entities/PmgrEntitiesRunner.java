package com.bst.application.pmgr.entities;

import java.util.Random;

import javax.persistence.EntityManager;
import javax.transaction.Transactional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Service;

import com.bst.pmgr.entities.Work;
import com.bst.pmgr.entities.repositories.WorkRepository;

@Service
@Transactional
public class PmgrEntitiesRunner implements CommandLineRunner {

	@Autowired
	private WorkRepository workRepository;

	@Autowired
	private EntityManager entityManager;

	@Override
	public void run(final String... args) throws Exception {

		final long countOfWork = Math.abs(new Random().nextLong()) % 10;
		long indexOfWork = 0;
		
		while (indexOfWork < countOfWork) {
			final Work work = new Work();
			this.workRepository.save(work);
			++indexOfWork;
		}

		this.entityManager.flush();
	}

}
