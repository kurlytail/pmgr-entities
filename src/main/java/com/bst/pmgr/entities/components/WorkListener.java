package com.bst.pmgr.entities.components;

import java.util.stream.Collectors;

import javax.persistence.PrePersist;

import org.kohsuke.randname.RandomNameGenerator;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import com.bst.pmgr.entities.Activity;
import com.bst.pmgr.entities.Document;
import com.bst.pmgr.entities.Process;
import com.bst.pmgr.entities.ProcessGroup;
import com.bst.pmgr.entities.Work;
import com.bst.pmgr.entities.repositories.ActivityRepository;
import com.bst.pmgr.entities.repositories.DocumentRepository;
import com.bst.pmgr.entities.repositories.ProcessGroupRepository;
import com.bst.pmgr.entities.repositories.ProcessRepository;
import com.bst.pmgr.entities.schema.SchemaService;

@Component
public class WorkListener {

	private final RandomNameGenerator rnd = new RandomNameGenerator(0);

	@Autowired
	private SchemaService schemaService;

	@Autowired
	private ActivityRepository activityRepository;

	@Autowired
	private DocumentRepository documentRepository;

	@Autowired
	private ProcessRepository processRepository;

	@Autowired
	private ProcessGroupRepository processGroupRepository;

	@PrePersist
	public void prePersist(final Work work) throws Exception {
		if (work.getName() == null) {
			work.setName(this.rnd.next());
		}

		if (work.getActivities().size() == 0) {
			work.getActivities().addAll(this.schemaService.getActivities().values().stream().map(metaActivity -> {
				final Activity activity = new Activity();
				activity.setMetaName(metaActivity.getName());
				activity.setName(this.rnd.next());
				activity.setWork(work);
				return activity;
			}).collect(Collectors.toList()));

			activityRepository.saveAll(work.getActivities());
		}

		if (work.getDocuments().size() == 0) {
			work.getDocuments().addAll(this.schemaService.getDocuments().values().stream().map(metaDocument -> {
				final Document document = new Document();
				document.setMetaName(metaDocument.getName());
				document.setName(this.rnd.next());
				document.setWork(work);
				return document;
			}).collect(Collectors.toList()));

			documentRepository.saveAll(work.getDocuments());
		}

		if (work.getProcesses().size() == 0) {
			work.getProcesses().addAll(this.schemaService.getProcesses().values().stream().map(metaProcess -> {
				final Process process = new Process();
				process.setMetaName(metaProcess.getName());
				process.setName(this.rnd.next());
				process.setWork(work);
				return process;
			}).collect(Collectors.toList()));

			processRepository.saveAll(work.getProcesses());
		}

		if (work.getProcessGroups().size() == 0) {
			work.getProcessGroups()
					.addAll(this.schemaService.getProcessGroups().values().stream().map(metaProcessGroup -> {
						final ProcessGroup processGroup = new ProcessGroup();
						processGroup.setMetaName(metaProcessGroup.getName());
						processGroup.setName(this.rnd.next());
						processGroup.setWork(work);
						return processGroup;
					}).collect(Collectors.toList()));

			processGroupRepository.saveAll(work.getProcessGroups());
		}
	}

}
