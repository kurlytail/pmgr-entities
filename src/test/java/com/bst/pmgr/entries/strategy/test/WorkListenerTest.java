package com.bst.pmgr.entries.strategy.test;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.autoconfigure.domain.EntityScan;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.context.ConfigurableApplicationContext;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
import org.springframework.test.context.TestExecutionListeners;
import org.springframework.test.context.TestExecutionListeners.MergeMode;
import org.springframework.test.context.junit.jupiter.SpringExtension;

import com.bst.pmgr.entities.Work;
import com.bst.pmgr.entities.repositories.WorkRepository;
import com.bst.pmgr.entities.schema.SchemaService;
import com.bst.pmgr.entities.strategy.WorkListener;
import com.bst.utility.components.AuditService;
import com.bst.utility.testlib.SnapshotListener;

@ExtendWith(SpringExtension.class)
@TestExecutionListeners(listeners = SnapshotListener.class, mergeMode = MergeMode.MERGE_WITH_DEFAULTS)
@DataJpaTest
// @ContextConfiguration(classes = { ComponentTestApplication.class })
public class WorkListenerTest {

	@TestConfiguration
	@EntityScan("com.bst.pmgr.entities")
	@EnableJpaRepositories("com.bst.pmgr.entities.repositories")
	static class WorkListenerTestConfiguration {
		@Bean
		public AuditService getAuditService() {
			return new AuditService();
		}

		@Bean
		public WorkListener getWorkListener() {
			return new WorkListener();
		}
		
		@Bean
		public SchemaService getSchemaService() {
			return new SchemaService();
		}

	}

	@Autowired
	private WorkRepository workRepository;
	
	@Autowired
	private WorkListener workListener;

	@Autowired
	private AuditService auditService;
	
	@Test
	public void testWorkListenerInitializeWork() throws Exception {
		Work work = new Work();
		work = this.workRepository.save(work);
		SnapshotListener.expect(work).toMatchSnapshot();
	}
}
