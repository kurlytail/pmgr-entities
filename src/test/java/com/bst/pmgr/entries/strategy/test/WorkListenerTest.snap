{
  "testWorkListenerInitializeWork" : [ {
    "id" : 1,
    "name" : "constructive_carrot",
    "activities" : [ {
      "id" : 2,
      "metaName" : "contractAdministration",
      "name" : "flexible_designer",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 3,
      "metaName" : "costEstimating",
      "name" : "linear_fund",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 4,
      "metaName" : "humanResourcePlanning",
      "name" : "popular_leaf",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 5,
      "metaName" : "qualitativeRiskAnalysis",
      "name" : "steady_parent",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 6,
      "metaName" : "planContracting",
      "name" : "abstract_rest",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 7,
      "metaName" : "createWbs",
      "name" : "controversial_supply",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 8,
      "metaName" : "directAndManageProjectExecution",
      "name" : "fragrant_absorption",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 9,
      "metaName" : "monitorAndControlProjectWork",
      "name" : "lively_cassette",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 10,
      "metaName" : "performanceReporting",
      "name" : "powerful_destruction",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 11,
      "metaName" : "riskManagementPlanning",
      "name" : "straight_fuss",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 12,
      "metaName" : "costBudgeting",
      "name" : "accessible_lecture",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 13,
      "metaName" : "quantitativeRiskAnalysis",
      "name" : "cooperative_participant",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 14,
      "metaName" : "acquireProjectTeam",
      "name" : "fresh_restriction",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 15,
      "metaName" : "communicationsPlanning",
      "name" : "loose_surgery",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 16,
      "metaName" : "activitySequencing",
      "name" : "pregnant_acceptance",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 17,
      "metaName" : "scheduleDevelopment",
      "name" : "strong_cat",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 18,
      "metaName" : "contractClosure",
      "name" : "addicted_deviation",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 19,
      "metaName" : "performQualityAssurance",
      "name" : "cruel_game",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 20,
      "metaName" : "requestSellerResponses",
      "name" : "full-time_legend",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 21,
      "metaName" : "developProjectCharter",
      "name" : "loyal_party",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 22,
      "metaName" : "manageStakeholders",
      "name" : "primary_retreat",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 23,
      "metaName" : "developPreliminaryProjectScopeStatement",
      "name" : "stupid_survivor",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 24,
      "metaName" : "planPurchasesAndAcquisitions",
      "name" : "aggressive_accumulation",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 25,
      "metaName" : "riskMonitoringAndControl",
      "name" : "cute_cattle",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 26,
      "metaName" : "closeProject",
      "name" : "generous_dialogue",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 27,
      "metaName" : "scopeControl",
      "name" : "major_garden",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 28,
      "metaName" : "integratedChangeControl",
      "name" : "professional_lemon",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 29,
      "metaName" : "riskResponsePlanning",
      "name" : "sudden_passion",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 30,
      "metaName" : "activityResourceEstimating",
      "name" : "alive_revival",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 31,
      "metaName" : "activityDurationEstimating",
      "name" : "dead_sweet",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 32,
      "metaName" : "manageProjectTeam",
      "name" : "glad_acquisition",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 33,
      "metaName" : "scopeVerification",
      "name" : "mathematical_celebration",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 34,
      "metaName" : "costControl",
      "name" : "proper_diet",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 35,
      "metaName" : "scopePlanning",
      "name" : "surprised_gear",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 36,
      "metaName" : "selectSellers",
      "name" : "ambitious_liability",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 37,
      "metaName" : "developProjectTeam",
      "name" : "decorative_pasture",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 38,
      "metaName" : "scheduleControl",
      "name" : "grand_rhythm",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 39,
      "metaName" : "scopeDefinition",
      "name" : "mechanical_syllable",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 40,
      "metaName" : "developProjectManagementPlan",
      "name" : "public_addition",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 41,
      "metaName" : "informationDistribution",
      "name" : "systematic_censorship",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 42,
      "metaName" : "activityDefinition",
      "name" : "anonymous_dimension",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 43,
      "metaName" : "qualityPlanning",
      "name" : "democratic_gentleman",
      "work" : 1,
      "tools" : [ ]
    }, {
      "id" : 44,
      "metaName" : "riskIdentification",
      "name" : "green_lid",
      "work" : 1,
      "tools" : [ ]
    } ],
    "documents" : [ {
      "id" : 45,
      "metaName" : "staffingManagementPlan",
      "name" : "middle-class_patience",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 46,
      "metaName" : "enterpriseEnvironmentalFactors",
      "name" : "quiet_rider",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 47,
      "metaName" : "selectedSellers",
      "name" : "tasty_syndrome",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 48,
      "metaName" : "validatedDefectRepair",
      "name" : "archaeological_admission",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 49,
      "metaName" : "resolvedIssues",
      "name" : "difficult_ceremony",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 50,
      "metaName" : "wbs",
      "name" : "hard_director",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 51,
      "metaName" : "forecastedCompletion",
      "name" : "minor_giant",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 52,
      "metaName" : "projectCalendar",
      "name" : "rational_lift",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 53,
      "metaName" : "administrativeClosureProcedure",
      "name" : "tense_pause",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 54,
      "metaName" : "procurementManagementPlan",
      "name" : "ashamed_ring",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 55,
      "metaName" : "qualityManagementPlan",
      "name" : "dirty_tail",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 56,
      "metaName" : "deliverables",
      "name" : "heavy_advantage",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 57,
      "metaName" : "riskManagementPlan",
      "name" : "modest_chalk",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 58,
      "metaName" : "qualityMetrics",
      "name" : "realistic_disagreement",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 59,
      "metaName" : "performanceMeasurementBaseline",
      "name" : "thirsty_glance",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 60,
      "metaName" : "implementedPreventiveActions",
      "name" : "attractive_limitation",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 61,
      "metaName" : "costBaseline",
      "name" : "domestic_peanut",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 62,
      "metaName" : "scopeBaseline",
      "name" : "hilarious_ritual",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 63,
      "metaName" : "projectScopeStatement",
      "name" : "moral_tape",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 64,
      "metaName" : "approvedDefectRepair",
      "name" : "regular_adviser",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 65,
      "metaName" : "resourceAvailability",
      "name" : "tired_chance",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 66,
      "metaName" : "wbsDictionary",
      "name" : "aware_discount",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 67,
      "metaName" : "qualityChecklists",
      "name" : "due_glory",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 68,
      "metaName" : "workPerformanceInformation",
      "name" : "honest_lion",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 69,
      "metaName" : "riskRegister",
      "name" : "musical_penalty",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 70,
      "metaName" : "rejectedChangeRequests",
      "name" : "reliable_robot",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 71,
      "metaName" : "contract",
      "name" : "traditional_taxi",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 72,
      "metaName" : "scheduleBaseline",
      "name" : "balanced_age",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 73,
      "metaName" : "acceptedDeliverables",
      "name" : "early_chapter",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 74,
      "metaName" : "implementedCorrectiveActions",
      "name" : "hot_disease",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 75,
      "metaName" : "milestoneList",
      "name" : "native_goal",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 76,
      "metaName" : "resourceRequirements",
      "name" : "residential_literacy",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 77,
      "metaName" : "projectScopeManagementPlan",
      "name" : "true_pensioner",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 78,
      "metaName" : "performanceReports",
      "name" : "beautiful_roof",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 79,
      "metaName" : "preliminaryProjectScopeStatement",
      "name" : "effective_tear",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 80,
      "metaName" : "approvedPreventiveActions",
      "name" : "ignorant_agony",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 81,
      "metaName" : "outsourceContract",
      "name" : "neutral_charity",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 82,
      "metaName" : "procurementDocumentPackage",
      "name" : "restricted_disorder",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 83,
      "metaName" : "projectManagementProcesses",
      "name" : "unanimous_gold",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 84,
      "metaName" : "proposals",
      "name" : "blank_loan",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 85,
      "metaName" : "activityCostEstimates",
      "name" : "eligible_performance",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 86,
      "metaName" : "recommendedCorrectiveActions",
      "name" : "implicit_rose",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 87,
      "metaName" : "forecasts",
      "name" : "noisy_telephone",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 88,
      "metaName" : "resourceCalendars",
      "name" : "right_aids",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 89,
      "metaName" : "qualityControlMeasurements",
      "name" : "unemployed_chauvinist",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 90,
      "metaName" : "scheduleManagementPlan",
      "name" : "bold_distributor",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 91,
      "metaName" : "projectOrganizationalCharts",
      "name" : "empty_gown",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 92,
      "metaName" : "closedContracts",
      "name" : "inadequate_log",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 93,
      "metaName" : "processImprovementPlan",
      "name" : "obese_person",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 94,
      "metaName" : "evaluationCriteria",
      "name" : "round_routine",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 95,
      "metaName" : "projectScheduleNetworkDiagrams",
      "name" : "uniform_temptation",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 96,
      "metaName" : "finalProductServiceOrResult",
      "name" : "bureaucratic_aisle",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 97,
      "metaName" : "activityDurationEstimates",
      "name" : "environmental_chemistry",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 98,
      "metaName" : "approvedChangeRequests",
      "name" : "incredible_division",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 99,
      "metaName" : "projectFundingRequirements",
      "name" : "occupational_grain",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 100,
      "metaName" : "makeBuyDecisions",
      "name" : "safe_loss",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 101,
      "metaName" : "requestedChanges",
      "name" : "unlike_phenomenon",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 102,
      "metaName" : "projectSchedule",
      "name" : "cautious_ruin",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 103,
      "metaName" : "activityAttributes",
      "name" : "eternal_tent",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 104,
      "metaName" : "organizationalProcessAssets",
      "name" : "indoor_allocation",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 105,
      "metaName" : "rolesAndResponsibilities",
      "name" : "old_chicken",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 106,
      "metaName" : "activityList",
      "name" : "seasonal_dog",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 107,
      "metaName" : "scheduleModelData",
      "name" : "useful_grant",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 108,
      "metaName" : "contractClosureProcedure",
      "name" : "charismatic_loyalty",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 109,
      "metaName" : "qualityBaseline",
      "name" : "exact_photography",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 110,
      "metaName" : "projectStatementOfWork",
      "name" : "influential_rush",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 111,
      "metaName" : "costManagementPlan",
      "name" : "optimistic_terrorist",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 112,
      "metaName" : "implementedDefectRepair",
      "name" : "senior_aluminium",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 113,
      "metaName" : "projectStaffAssignments",
      "name" : "vague_chimney",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 114,
      "metaName" : "teamPerformanceAssessment",
      "name" : "chronic_domination",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 115,
      "metaName" : "resourcesBreakdownStructure",
      "name" : "exotic_gravel",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 116,
      "metaName" : "recommendedDefectRepair",
      "name" : "insufficient_machine",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 117,
      "metaName" : "projectCharter",
      "name" : "organic_pie",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 118,
      "metaName" : "contractDocumentation",
      "name" : "sexual_sailor",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 119,
      "metaName" : "projectManagementPlan",
      "name" : "varied_thanks",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 120,
      "metaName" : "approvedCorrectiveActions",
      "name" : "clean_amendment",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 121,
      "metaName" : "costEstimate",
      "name" : "experimental_chord",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 122,
      "metaName" : "activityResourceRequirements",
      "name" : "interactive_double",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 123,
      "metaName" : "procurementDocuments",
      "name" : "outer_grief",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 124,
      "metaName" : "recommendedPreventiveActions",
      "name" : "shy_maid",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 125,
      "metaName" : "contractManagementPlan",
      "name" : "vicious_pigeon",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 126,
      "metaName" : "performanceMeasurements",
      "name" : "cold_salt",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 127,
      "metaName" : "riskRelatedContractualAgreements",
      "name" : "extinct_theorist",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 128,
      "metaName" : "qualifiedSellerList",
      "name" : "international_angel",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 129,
      "metaName" : "activityCostEstimateSupportingDetail",
      "name" : "paralyzed_cinema",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 130,
      "metaName" : "communicationsManagementPlan",
      "name" : "skilled_draft",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 131,
      "metaName" : "implementedChangeRequests",
      "name" : "visual_grounds",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    }, {
      "id" : 132,
      "metaName" : "contractStatementOfWork",
      "name" : "commercial_makeup",
      "work" : 1,
      "tools" : [ ],
      "sections" : [ ]
    } ],
    "tools" : [ ],
    "processes" : [ {
      "id" : 138,
      "name" : "false_pin",
      "metaName" : "communicationsManagement",
      "work" : 1
    }, {
      "id" : 139,
      "name" : "joint_sand",
      "metaName" : "timeManagement",
      "work" : 1
    }, {
      "id" : 140,
      "name" : "passionate_thesis",
      "metaName" : "qualityManagement",
      "work" : 1
    }, {
      "id" : 141,
      "name" : "smart_ankle",
      "metaName" : "scopeManagement",
      "work" : 1
    }, {
      "id" : 142,
      "name" : "warm_citizen",
      "metaName" : "costManagement",
      "work" : 1
    }, {
      "id" : 143,
      "name" : "competent_drawing",
      "metaName" : "integrationManagement",
      "work" : 1
    }, {
      "id" : 144,
      "name" : "fast_guest",
      "metaName" : "procurementManagement",
      "work" : 1
    }, {
      "id" : 145,
      "name" : "kind_manner",
      "metaName" : "humanResourceManagement",
      "work" : 1
    }, {
      "id" : 146,
      "name" : "peaceful_pitch",
      "metaName" : "riskManagement",
      "work" : 1
    } ],
    "processGroups" : [ {
      "id" : 133,
      "name" : "soft_sauce",
      "metaName" : "monitoringAndControlling",
      "work" : 1
    }, {
      "id" : 134,
      "name" : "well_thread",
      "metaName" : "planning",
      "work" : 1
    }, {
      "id" : 135,
      "name" : "comprehensive_anticipation",
      "metaName" : "closing",
      "work" : 1
    }, {
      "id" : 136,
      "name" : "federal_clash",
      "metaName" : "initiating",
      "work" : 1
    }, {
      "id" : 137,
      "name" : "latest_drill",
      "metaName" : "executing",
      "work" : 1
    }, 133, 134, 135, 136, 137 ]
  } ]
}